### PR TITLE
fix for function app, app setting for storage account

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
https://github.com/Azure/azure-libraries-for-net/issues/485
If app service plan is not consumption, skip WEBSITE_CONTENTAZUREFILECONNECTIONSTRING and WEBSITE_CONTENTSHARE

https://github.com/Azure/azure-libraries-for-net/issues/457
If it is Update (NOT Create) operation, and storage account is not set, do not create new storage account (and do not update related app setting).

### CODE REVIEW

File: src/ResourceManagement/AppService/FunctionAppImpl.cs [hunk 1]
```diff
@@ -154,17 +160,31 @@ internal override async Task<Models.SiteInner> SubmitAppSettingsAsync(SiteInner
             }
         }
 
+        private bool IsConsumptionAppServicePlan(PricingTier pricingTier)
+        {
+            SkuDescription description = pricingTier.SkuDescription;
```
Comment (yaohaizh) [inline src/ResourceManagement/AppService/FunctionAppImpl.cs]: Null check

File: src/ResourceManagement/AppService/FunctionAppImpl.cs [hunk 2]
```diff
@@ -137,13 +137,19 @@ internal override async Task<Models.SiteInner> SubmitAppSettingsAsync(SiteInner
...
+                bool isConsumptionPlan = IsConsumptionAppServicePlan(servicePlan.PricingTier);
+
                 var keys = await storageAccountToSet.GetKeysAsync(cancellationToken);
                 var connectionString = string.Format("DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}",
                     storageAccountToSet.Name, keys[0].Value);
                 WithAppSetting("AzureWebJobsStorage", connectionString);
                 WithAppSetting("AzureWebJobsDashboard", connectionString);
-                WithAppSetting("WEBSITE_CONTENTAZUREFILECONNECTIONSTRING", connectionString);
-                WithAppSetting("WEBSITE_CONTENTSHARE", SdkContext.RandomResourceName(Name, 32));
+                if (isConsumptionPlan)
```
Comment (yaohaizh) [inline src/ResourceManagement/AppService/FunctionAppImpl.cs]: Doesn't need additional variable here.

File: src/ResourceManagement/AppService/FunctionAppImpl.cs [hunk 3]
```diff
@@ -154,17 +160,31 @@ internal override async Task<Models.SiteInner> SubmitAppSettingsAsync(SiteInner
...
+                {
+                    WithNewConsumptionPlan();
+                }
+                if (currentStorageAccount == null && storageAccountToSet == null && storageAccountCreatable == null)
+                {
+                    WithNewStorageAccount(SdkContext.RandomResourceName(Name, 20).Replace("-", String.Empty), Storage.Fluent.Models.SkuName.StandardGRS);
+                }
+            } else
             {
-                WithNewStorageAccount(SdkContext.RandomResourceName(Name, 20).Replace("-", String.Empty), Storage.Fluent.Models.SkuName.StandardGRS);
```
Comment (yaohaizh) [inline src/ResourceManagement/AppService/FunctionAppImpl.cs]: The else is not necessary. Please remove it to keep the code clean.

File: src/ResourceManagement/AppService/FunctionAppImpl.cs [hunk 4]
```diff
@@ -154,17 +160,31 @@ internal override async Task<Models.SiteInner> SubmitAppSettingsAsync(SiteInner
...
+                // for Update+Apply, skip check null and create new app service plan and storage account, use existing ones if caller does not specify otherwise
             }
-            return await base.CreateAsync(cancellationToken);
+            return await base.CreateAsync(cancellationToken, multiThreaded);
```
Comment (yaohaizh) [inline src/ResourceManagement/AppService/FunctionAppImpl.cs]: What's the reason to add the multiThreaded here?

Comment (weidongxu-microsoft) [reply src/ResourceManagement/AppService/FunctionAppImpl.cs]: method signature is
`public override async Task<IFunctionApp> CreateAsync(CancellationToken cancellationToken = default(CancellationToken), bool multiThreaded = true)`

so I guess previous base call miss that multiThreaded param.

**Pull request merged**